src/js_realm.cpp

```diff
@@ -249,11 +249,28 @@ JSValueRef RealmGetProperty(JSContextRef ctx, JSObjectRef object, JSStringRef pr
     return NULL;
 }
 
+std::string RealmObjectTypeForValue(SharedRealm &realm, JSContextRef ctx, JSValueRef value) {
+    if (JSValueIsObject(ctx, value) && JSObjectIsConstructor(ctx, (JSObjectRef)value)) {
```

Comment:
`JSObjectIsConstructor` will return `true` if the function can be called with `new`, which basically only excludes arrow functions and native functions, so I'll update the schema parsing code to remove the `JSObjectIsFunction` check.